### PR TITLE
feat: add support for docs

### PR DESCRIPTION
**Pull request by vsavkin**

### CODE REVIEW

File: libs/feature-generate/src/lib/schematic/schematic.component.ts [hunk 1]
```diff
@@ -221,6 +236,37 @@ export class SchematicComponent implements OnInit {
...
+            }
+          })
+        )
+        .subscribe(r => {
+          if (!r) {
+            this.docs = [];
+          } else {
+            this.docs = r.data.workspace.docs.schematicDocs;
+          }
+          console.log('schematicDocs', this.docs);
```
Comment (mrmeku) [inline libs/feature-generate/src/lib/schematic/schematic.component.ts]: Remove the console log

File: libs/feature-run/src/lib/target/target.component.ts [hunk 2]
```diff
@@ -132,6 +139,32 @@ export class TargetComponent implements OnInit {
...
+            }
+          })
+        )
+        .subscribe(r => {
+          if (!r) {
+            this.docs = [];
+          } else {
+            this.docs = r.data.workspace.docs.schematicDocs;
+          }
+          console.log('targetDocs', this.docs);
```
Comment (mrmeku) [inline libs/feature-run/src/lib/target/target.component.ts]: Remove the console log

File: libs/feature-workspaces/src/lib/projects/projects.component.ts [hunk 3]
```diff
@@ -48,6 +57,21 @@ export class ProjectsComponent implements OnInit {
         return { ...w, projects };
       })
     );
+
+    if (this.settings.showDocs) {
+      this.docsSubscription = this.route.params
+        .pipe(switchMap(p => this.workspaceDocsGQL.fetch({ path: p.path })))
+        .subscribe(r => {
+          this.docs = r.data.workspace.docs.workspaceDocs;
+          console.log('workspaceDocs', this.docs);
```
Comment (mrmeku) [inline libs/feature-workspaces/src/lib/projects/projects.component.ts]: Remove the console log

File: libs/feature-workspaces/src/lib/projects/projects.component.ts [hunk 4]
```diff
@@ -48,6 +57,21 @@ export class ProjectsComponent implements OnInit {
         return { ...w, projects };
       })
     );
+
+    if (this.settings.showDocs) {
+      this.docsSubscription = this.route.params
+        .pipe(switchMap(p => this.workspaceDocsGQL.fetch({ path: p.path })))
+        .subscribe(r => {
```
Comment (mrmeku) [inline libs/feature-workspaces/src/lib/projects/projects.component.ts]: We should set this as a observable as a class member and subscribe within the template

File: server/src/api/docs.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+      p.schematicDocs(collectionName, collectionVersion, name)
+    );
+  }
+
+  openDoc(id: string): Observable<boolean> {
+    return of(...this.providers).pipe(
+      concatMap(p =>
+        p.openDoc(id).pipe(
+          catchError(e => {
+            console.log(`error`, e.message);
```
Comment (mrmeku) [inline server/src/api/docs.ts]: Add to analytics?

File: server/src/api/docs.ts [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+  private collectInfoFromProviders(
+    callback: (d: DocsProvider) => Observable<Doc[]>
+  ): Observable<Doc[]> {
+    return of(...this.providers).pipe(
+      concatMap(d => {
+        try {
+          return callback(d).pipe(
+            catchError(e => {
+              console.log(`error`, e.message);
```
Comment (mrmeku) [inline server/src/api/docs.ts]: Analytics?

File: server/src/server.ts [hunk 7]
```diff
@@ -40,8 +44,11 @@ app.use(express.static(path.join(__dirname, 'public')));
 
 export function start(port: number) {
   app.listen(port ? port : 7777);
-}
 
-if (process.argv[2]) {
-  start(+process.argv[2]);
+  // install plugins
+  console.log('starting----');
```
Comment (mrmeku) [inline server/src/server.ts]: Do you mean to have this duplicated 4 times?

Comment (vsavkin) [reply server/src/server.ts]: Removed it

File: server/src/api/docs.ts (hunk 5 above)
Comment (vsavkin) [reply server/src/api/docs.ts]: Currently, we don't report exceptions to analytics. I will update all of them in a single PR.

File: server/src/api/docs.ts (hunk 6 above)
Comment (vsavkin) [reply server/src/api/docs.ts]: Currently, we don't report exceptions to analytics. I will update all of them in a single PR.

**Pull request merged**